### ISSUE TITLE
Untrusted devices notice doesn't reappar over time when new devices present thus causing problems on the receivers' end

### ISSUE THREAD

**Issue by lytoric** [poljar/weechat-matrix#62]
I've had an occurrence of an issue where after about 2-3 days of having weechat-matrix loaded and connected to a network, the `(Untrusted devices found in room...)` notice would no longer appear when new untrusted devices were present, thus causing any messages sent from inside weechat-matrix to be unreadable to the users with these untrusted devices, and causing them to receive this message:
```
** Unable to decrypt: The sender's device has not sent us the keys for this message. **
Re-request encryption keys from your other devices.
```
Reloading the script caused it to function as intended again.

**Comment by poljar** [comment]
That's weird. It could happen if the server for some reason doesn't tell us that the device list changed or some subtle bug in the way we handle that. Will check the code again.

Reloading the script causes the full device list to be queried so it's normal that it fixed it.

Thanks for the report.

**Comment by poljar** [comment]
Ok, I think I figured out the issue. Well at least one reason why this might happen, short story: we have flags that remember if we made a request to not make another one while we wait on a response.

Those flags weren't properly cleared on disconnects and we end up in a state where we don't make any key queries because the flag is stuck.

Thus we never see a new device even though the lib correctly tells us to do a key query.

This should be fixed now with c1cfc4a8e4a67989d526a2c9142575e447f88cd7.
